### PR TITLE
#issues-1645 Byzer-lang variables  with session scope fails to evaluate

### PR DESCRIPTION
**Pull request by anan0120**
# What changes were proposed in this pull request?

check [ISSUE-1645](https://github.com/byzer-org/byzer-lang/issues/1645)

# How was this patch tested?

20_nesting_variables_across_cells_test.mlsql

# Are there and DOC need to update?

# Spark Core Compatibility

### CODE REVIEW

File: streamingpro-it/src/test/resources/sql/simple/20_nesting_variables_across_cells_test.mlsql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+set table_name="T_TableName" where scope="session";
+set ods_table_name="ODS_${table_name}" where scope="session";
+select "${ods_table_name}" as name as `${ods_table_name}`;
+
+-- second cell
+
+select name from `${ods_table_name}` as ccc;
+
+-- make sure status of  all models are success.
+!assert result ''':name=="ODS_T_TableName"'''  "all model status should be success";
```
Comment (allwefantasy) [inline streamingpro-it/src/test/resources/sql/simple/20_nesting_variables_across_cells_test.mlsql]: This line equals:

assert(:name=="ODS_T_Table_Name", "The variable ods_table_name should be ODS_T_TableName")

So it's better to change the description in the last part.

**Review by allwefantasy — CHANGES_REQUESTED** [review]

**Comment by anan0120** [comment]
Corrected

File: streamingpro-it/src/test/resources/sql/simple/20_nesting_variables_across_cells_test.mlsql (hunk 1 above)
Comment (allwefantasy) [reply streamingpro-it/src/test/resources/sql/simple/20_nesting_variables_across_cells_test.mlsql]: 1) remove the comment line `--make sure satus....`;
2) change the description `all model status should be success` which is not suite for your test.

**Comment by anan0120** [comment]
update nesting_variables_across file describe

File: streamingpro-it/src/test/resources/sql/simple/20_nesting_variables_across_cells_test.mlsql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+set table_name="T_TableName" where scope="session";
+set ods_table_name="ODS_${table_name}" where scope="session";
+select "${ods_table_name}" as name as `${ods_table_name}`;
+
+-- second cell
+
+select name from `${ods_table_name}` as ccc;
+
+-- make sure The variable ods_table_name should be ODS_T_TableName.
+!assert result ''':name=="ODS_T_TableName"'''  "The variable ods_table_name should be ODS_T_TableName";
```
Comment (allwefantasy) [inline streamingpro-it/src/test/resources/sql/simple/20_nesting_variables_across_cells_test.mlsql:13]: The target table is not result

**Review by allwefantasy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**